libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamInput.java

```diff
@@ -750,6 +751,8 @@ public Object readGenericValue() throws IOException {
                 return readCollection(StreamInput::readGenericValue, HashSet::new, Collections.emptySet());
             case 26:
                 return readBigInteger();
+            case 27:
+                return readSemverRange();
```

Comment:
When we add support for multiple ranges, will this need to change or will we be adding a collection of semver ranges? Meaning does it make sense to make this a collection (of 1) right now?

Reply:
This will not need to change when we enable support for multiple ranges. The PluginInfo serialization process will write a [list of SemverRange](https://github.com/opensearch-project/OpenSearch/pull/11441/files#diff-9706884d125350e82403647f631ce066f4aa2e99747c4247f3dd943197db6fc4R211) type objects. [Every object in the list](https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamOutput.java#L702) will be [serialized as SemverType](https://github.com/opensearch-project/OpenSearch/pull/11441/files#diff-44bd9f0f10ddc587fabe125f40ee3889e5a1fa9c56b78440f41fd147df1df6bfR788) object. 
Deserialization process of PluginInfo [will read](https://github.com/opensearch-project/OpenSearch/pull/11441/files#diff-9706884d125350e82403647f631ce066f4aa2e99747c4247f3dd943197db6fc4R194) [a list of](https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamInput.java#L716) [SemverRange](https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamInput.java#L782) type objects. This has been tested in a previous revision when support for multiple ranges was added.